frontend/public/components/cluster-settings/cluster-settings.tsx

```diff
@@ -44,71 +56,91 @@ const getClusterUpdateStatus = (cv: K8sResourceKind): ClusterUpdateStatus => {
   return _.isEmpty(updates) ? ClusterUpdateStatus.UpToDate : ClusterUpdateStatus.UpdatesAvailable;
 };
 
-const getIconClass = (status: ClusterUpdateStatus) => {
+const getIcon = (status: ClusterUpdateStatus) => {
   return {
-    [ClusterUpdateStatus.UpToDate]: 'pficon pficon-ok text-success',
-    [ClusterUpdateStatus.UpdatesAvailable]: 'fa fa-arrow-circle-up text-success',
-    [ClusterUpdateStatus.Updating]: 'pficon pficon-in-progress',
-    [ClusterUpdateStatus.Failing]: 'pficon pficon-error-circle-o text-danger',
-    [ClusterUpdateStatus.ErrorRetrieving]: 'pficon pficon-error-circle-o text-danger',
+    [ClusterUpdateStatus.UpToDate]: <Icon type="pf" name="ok" />,
```

Comment:
I think we should avoid the inclusion of additional `<Icon>` instances because they're adding to our dependency overhead for very little gain.  The majority of of icons are straight jsx.

Reply:
There are a few things I like about using these components:

1. WIth the pf icons, they will be colored accordingly and I don't need to add extra css classes to the ok, error, or warning icons.
2. I always forget to add the `aria-hidden` attribute when creating icons, which is rolled into the Icon components.
3. They abstract the css class names and make the code more legible.
4. It's DRY. We are creating a lot of the same markup over and over for icons, and the components alleviate that repitition. Less room for error.

Reply:
> WIth the pf icons, they will be colored accordingly and I don't need to add extra CSS classes to the ok, error, or warning icons.

No extra classes should be required.  The color should be part of `.pficon-*`.

I don't disagree with your other points.  But I do disagree with using the component in some of the cases but not all.  I'd rather see us do thing consistently across the board.  One of my mantras:  "It's better to be consistently wrong than inconsistently right".

Reply:
Yeah, I agree we should be consistent with either using them or not. Since this is outside of the scope of this PR, I'll just go ahead and take those out and we can kick that it down the road a bit.